pkg/crosscluster/logical/replication_statements_test.go

```diff
@@ -47,6 +52,34 @@ func TestReplicationStatements(t *testing.T) {
 		)
 	}
 
+	prepareStatement := func(t *testing.T, db *gosql.DB, columnTypes []*types.T, statement statements.Statement[tree.Statement]) {
```

Comment:
out of curiosity, why was this change necessary?

Reply:
This change is necessary because the query needs to know the type of every parameter and there is no way for sql execution to deduce the type of a parameter that does not show up in the query anywhere.

This is relevant because the crud writer passes around a single tree.Datums for each row, but not every query needs every column.